src/lib/code-analyzer-run-action.ts

```diff
@@ -112,4 +116,28 @@ export class CodeAnalyzerRunAction {
         }
         this.display.displayInfo(messages.info.finishedScan(numFilesScanned, filesWithViolations.size, violations.length));
     }
+
+    /**
+     * An engine won't start, and we want to limit how many times a user has to encounter this error.
+     * If the user has seen the error for this engine in this session - don't show it again.
+     * If it's the first time seeing it, then set the workspace state key and notify the user.
+     */
+    private handleEngineError(engine: string) {
+        // Use a key with date component to reset warnings daily
+        const today = getCurrentDate();
+        const engineWorkspaceKey = `${Constants.ENGINE_WARNING_PREFIX}${engine}_${today}`;
+        const seenThisEngineError = WorkspaceState.getValue<boolean>(engineWorkspaceKey) ?? false;
+
+        if (!seenThisEngineError) {
+            WorkspaceState.setValue(engineWorkspaceKey, true);
```

Comment:
After creating this PR, I'm realizing this could be a good spot for some telemetry. Thoughts?